core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -479,16 +480,45 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     }
   }
 
-  // Assures the mUri is the same as the master address in the current client context.
-  private void checkMasterAddress() throws IOException {
-    InetSocketAddress masterAddress = ClientContext.getMasterAddress();
+  /**
+   * @return true if the new context is the same as the existing file system context
+   */
+  private boolean checkFileSystemContext() {
+    InetSocketAddress masterAddress = sFileSystemContext.getMasterAddress();
+    Subject subject = sFileSystemContext.getParentSubject();
+    Subject newSubject = getHadoopSubject();
     boolean sameHost = masterAddress.getHostString().equals(mUri.getHost());
     boolean samePort = masterAddress.getPort() == mUri.getPort();
-    if (sameHost && samePort) {
-      return;
+    boolean sameSubject = false;
+    if (subject == null) {
+      sameSubject = newSubject == null;
+    } else {
+      sameSubject = subject.equals(newSubject);
     }
+    return sameHost && samePort && sameSubject;
+    /*
     throw new IOException(ExceptionMessage.DIFFERENT_MASTER_ADDRESS
         .getMessage(mUri.getHost() + ":" + mUri.getPort(), masterAddress));
+        */
+  }
+
+  /**
+   * @return the hadoop subject if exists
+   */
+  private Subject getHadoopSubject() {
+    try {
+      UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
```

Comment:
is it possible for the current user to be different from the real user? when superuser proxy is used in UGI. Shall we use `UserGroupInformation.getRealUser()` in some cases? 

https://hadoop.apache.org/docs/r2.7.2/hadoop-project-dist/hadoop-common/Superusers.html